CHANGELOG.md

```diff
@@ -10,12 +10,289 @@ New features:
 
 Bugfixes:
 
-* Only include direct dependencies in the output for `purs graph` (#3993, @colinwahl)
+Other improvements:
 
-Fixes a bug where the transitive closure of a module's dependencies
-where included in the `depends` field in the output for `purs graph`.
+## v0.14.0
 
-Other improvements:
+### Polykinds
+
+Polymorphic kinds, based on the [Kind Inference for Datatypes](https://richarde.dev/papers/2020/kind-inference/kind-inference.pdf) paper (#3779, #3831, #3929, @natefaubion)
+
+Kinds are to types what types are to terms (although kinds are themselves types now), but whereas we have polymorphic types, kinds were monomorphic.
```

Comment:
```suggestion
Conceptually, kinds are to types what types are to terms (although kinds are themselves types now). But while we have polymorphic types, kinds were previously monomorphic.
```

I found this sentence a bit difficult to read; I think it's a little clearer to separate the comment about kinds -> types and types -> terms from the comment about polymorphism.

Reply:
I'm unsure about the addition of "Conceptually" here. Did you intend to imply that in actuality, kinds are _not_ to types what types are to terms? I don't think the statement "kinds are to types what types are to terms" needs any qualification.

Reply:
I took the ‘conceptually’ to indicate that the analogy is imperfect (it was a better analogy when kinds were a distinct universe above types the same way that types are a distinct universe above terms, but now kinds only exist as a relationship between types). I'm fine with that but maybe ‘approximately’ would be a clearer shade of paint to you?

Reply:
I think if you understand "kind" as "a type which, in this context, is being used to classify a type rather than a term", then I don't see this as an approximation: isn't the relationship the same, even if types are differ from terms in other ways?

Reply:
I think anyone who already understands ‘kind’ to mean that doesn't need this sentence to clarify what kinds are. :smile: I figure this is just a brief summary targeting someone who doesn't have a crisp idea of what a kind is and wants to understand how/if this change affects them, and that person might be confused unless they understand that the analogy only holds for ‘kind-of’ the relationship relative to ‘type-of’ but not for ‘kind’ the category of data manipulated by the compiler relative to ‘type’ the category etc.

Reply:
Right - I mean it's a definition, basically, isn't it? Ignore the "if you understand 'kind' as ..." part, then. I think whenever you say "A is to B as C is to D" and A/C and B/D aren't exactly the same things there's always going to be at least one way in which they differ, so I'd say it's reasonable to consider the fact that types and kinds live in the same namespace now (whereas types and terms don't) immaterial to the original statement. But maybe we can avoid this issue by just being a bit more explicit to start with? "Just as types classify terms, kinds classify types."?